mesonbuild/wrap/wrap.py

```diff
@@ -720,6 +720,29 @@ def get_data(self, urlstring: str) -> T.Tuple[str, str]:
             resp = open_wrapdburl(urlstring, allow_insecure=self.allow_insecure, have_opt=self.wrap_frontend)
         elif WHITELIST_SUBDOMAIN in urlstring:
             raise WrapException(f'{urlstring} may be a WrapDB-impersonating URL')
+        elif url.scheme == 'sftp':
+            sftp = shutil.which('sftp')
+            if sftp is None:
+                raise WrapException('Scheme sftp is not available. Install sftp to enable it.')
+            with tempfile.TemporaryDirectory() as workdir, \
+                    tempfile.NamedTemporaryFile(mode='wb', dir=self.cachedir, delete=False) as tmpfile:
+                args = []
+                if 'source_hostkey' in self.wrap.values:
+                    with open(os.path.join(workdir, 'known_hosts'), 'w', encoding='utf-8') as f:
+                        f.write(f'[{url.hostname}]:{url.port} {self.wrap.get("source_hostkey")}')
+                    args += ['-o', 'UserKnownHostsFile=known_hosts']
+                if 'source_identityfile' in self.wrap.values:
+                    args += ['-o', f'IdentityFile={self.wrap.get("source_identityfile")}']
```

Comment:
Code that fiddles with the end user's crypto setup makes me especially anxious. If you set these arguments in the wrap file, then that file can only work on the machine it has been generated on (or en exact replica of it). There might also be some sort of a information leakage vulnerability here, but I don't have the skills to say so one way or another.

Because of these and similar reasons we try to stay away from access control as much as possible. Are these features absolutely necessary or can we just say "set up access control outside Meson"?

Reply:
The alternatives for testing the SSH part is authentication with either a key pair (which is generated in each run here), or a password (meaning we need to have a test user with a known password).
The intention was not that "source_identityfile" should be used outside of testing, as indicated by the commit message. If there's a better way to inject the necessary information into the code, for test purposes only, I'm open to suggestions.

Alternatives for the testing as a whole might be to:
a) skip SSH, just do the SFTP part (I haven't looked into whether this makes it possible to get rid of authentication), or
b) skip testing altogether.

Reply:
Sounds reasonable to me! Removed the test code.